### PR TITLE
Documentation: fixes for Sphinx 1.6

### PR DESCRIPTION
**Pull request by sbesson**
This PR includes a set of changes making our documentation compatible with the latest version of Sphinx:

- migrates the custom sidebars under `_templates` (see corresponding PRs in https://github.com/ome/ome-model/pull/36)
- fix the usage/references of the option markup across the code and defines `bfconvert -option`
- fix the table in the Micro-Manager user page (to prevent multiple row spannings) and add internal references

Additionally following https://github.com/openmicroscopy/bioformats/pull/2852#discussion_r118439140, this also increases the stringency of the Sphinx build to set `-W` and treat warnings as errors by default. This should apply to the Travis build in particular and let the Travis build fail for invalid documentation syntaxes.

Finally, db82d65 uncaps the version of Sphinx in `.travis.yml` to use a more recent and more stringent version. This should allow to test the validity this PR although we could cap this version in Travis.

All changes should be backwards compatible and build, deploy and render similarly to the latest release state with our CI builds - see https://www.openmicroscopy.org/site/support/bio-formats5.5-staging/.

### CODE REVIEW

**Comment by mtbc** [comment]
:+1: for increasing automated checking.

File: docs/sphinx/users/comlinetools/conversion.rst [hunk 1]
```diff
@@ -80,6 +80,21 @@ The output file format is determined by the extension of the output file, e.g.
     single tile with a few different tile sizes using the :option:`-crop`
     option. This gives an idea of what the most performant size will be.
 
+.. option:: -crop X,Y,WIDTH,HEIGHT
+
+    For very large images, it may also be useful to convert a small tile from
+    the image instead of reading everything into memory. To convert the
+    upper-left-most 512x512 tile from the images:
```
Comment (rleigh-codelibre) [inline docs/sphinx/users/comlinetools/conversion.rst]: 512×512 ?

File: docs/sphinx/developers/commit-testing.rst [hunk 2]
```diff
@@ -173,7 +173,7 @@ name of the method:
   cd /path/to/bioformats/components/formats-gpl/test/matlab
   runxunit TestBfsave:testLZW
 
-Finally to output the test results under XML format, you can use the :option:`-xmlfile` option:
+Finally to output the test results under XML format, you can use the ``-xmlfile`` option:
```
Comment (rleigh-codelibre) [inline docs/sphinx/developers/commit-testing.rst]: `Finally,` - add comma.

**Comment by rleigh-codelibre** [comment]
All looks fine; a couple of minor details noted.

**Comment by sbesson** [comment]
Last commit should fix the issue raised in the PR comments and review all usages of dimension separator symbols in the Sphinx documentation.

**Comment by rleigh-codelibre** [comment]
That all looks great, thanks.  Please merge when green.

**Pull request merged**